Public API regression with v0.12.13?

Hi there! Thanks a ton for maintaining `reqwest`.

I'm opening this up as a potential (maybe intentional?) public API change/regression between patch versions. If you determine that it was intentional and that the downstream (`reqwest-middleware` in this case) is responsible, then please feel free to close this and I'll pursue things further there 🙂 

## Problem

v0.12.13 appears to include #2573, which removes the `RequestBuilder::fetch_mode_no_cors()` helper on non-WASM targets.

This is understandable because it's a no-op elsewhere, but it changed the public API in such a way that's breaking `reqwest-middleware` here:

https://github.com/TrueLayer/reqwest-middleware/blob/101fa6371b8f90b7bf250050b9cecb7ed22ef9ab/reqwest-middleware/src/client.rs#L521-L535

For projects that users install with `cargo install` (where locking is unfortunately not the default), this produces the following error:

```
error[E0599]: no method named `fetch_mode_no_cors` found for struct `reqwest::RequestBuilder` in the current scope
   --> /Users/william/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/reqwest-middleware-0.4.1/src/client.rs:532:31
    |
532 |             inner: self.inner.fetch_mode_no_cors(),
    |                               ^^^^^^^^^^^^^^^^^^ method not found in `RequestBuilder`

   Compiling crossbeam-deque v0.8.6
For more information about this error, try `rustc --explain E0599`.
error: could not compile `reqwest-middleware` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

## Context

I maintain [zizmor](https://github.com/woodruffw/zizmor), and https://github.com/woodruffw/zizmor/issues/600 brought this to my attention -- zizmor depends on `reqwest` and `reqwest-middleware`, and the latter uses a public API that no longer exists in the latest version of the former.

This only happens to users who don't run `cargo install --locked` and therefore isn't a critical problem for me personally. However, I figured I'd raise it upstream anyways 🙂

--- a/src/async_impl/request.rs
+++ b/src/async_impl/request.rs
@@ -459,6 +459,19 @@ impl RequestBuilder {
         self
     }
 
+    // This was a shell only meant to help with rendered documentation.
+    // However, docs.rs can now show the docs for the wasm platforms, so this
+    // is no longer needed.
+    //
+    // You should not otherwise depend on this function. It's deprecation
+    // is just to nudge people to reduce breakage. It may be removed in a
+    // future patch version.
+    #[doc(hidden)]
+    #[cfg_attr(target_arch = "wasm32", deprecated)]
+    pub fn fetch_mode_no_cors(self) -> RequestBuilder {
+        self
+    }
+
     /// Build a `Request`, which can be inspected, modified and executed with
     /// `Client::execute()`.
     pub fn build(self) -> crate::Result<Request> {

